python_modules/libraries/dagster-gcp/dagster_gcp/types.py

```diff
@@ -0,0 +1,123 @@
+import re
+import six
+
+from google.cloud.bigquery.job import (
+    CreateDisposition,
+    Encoding,
+    SchemaUpdateOption,
+    SourceFormat,
+    QueryPriority,
+    WriteDisposition,
+)
+
+from dagster import DagsterUserError, Enum, EnumValue, ConfigScalar
+
+
+# Project names are permitted to have alphanumeric, dashes and underscores, up to 1024 characters.
+RE_PROJECT = r'[\w\d\-\_]{1,1024}'
+
+# Datasets and tables are permitted to have alphanumeric or underscores, no dashes allowed, up to
+# 1024 characters
+RE_DS_TABLE = r'[\w\d\_]{1,1024}'
+
+
+BQCreateDispositionCreateIfNeeded = EnumValue(CreateDisposition.CREATE_IF_NEEDED)
+BQCreateDispositionCreateNever = EnumValue(CreateDisposition.CREATE_NEVER)
+BQCreateDisposition = Enum(
+    name='BQCreateDisposition',
+    enum_values=[BQCreateDispositionCreateIfNeeded, BQCreateDispositionCreateNever],
+)
+
+BQPriorityBatch = EnumValue(QueryPriority.BATCH)
+BQPriorityInteractive = EnumValue(QueryPriority.INTERACTIVE)
+BQPriority = Enum(name='BQPriority', enum_values=[BQPriorityBatch, BQPriorityInteractive])
+
+BQSchemaUpdateOptionAllowFieldAddition = EnumValue(
+    SchemaUpdateOption.ALLOW_FIELD_ADDITION,
+    description='Allow adding a nullable field to the schema.',
+)
+BQSchemaUpdateOptionAllowFieldRelaxation = EnumValue(
+    SchemaUpdateOption.ALLOW_FIELD_RELAXATION,
+    description='Allow relaxing a required field in the original schema to nullable.',
+)
+BQSchemaUpdateOption = Enum(
+    name='BQSchemaUpdateOption',
+    enum_values=[BQSchemaUpdateOptionAllowFieldAddition, BQSchemaUpdateOptionAllowFieldRelaxation],
+)
+
+BQWriteDispositionWriteAppend = EnumValue(WriteDisposition.WRITE_APPEND)
+BQWriteDispositionWriteEmpty = EnumValue(WriteDisposition.WRITE_EMPTY)
+BQWriteDispositionWriteTruncate = EnumValue(WriteDisposition.WRITE_TRUNCATE)
+BQWriteDisposition = Enum(
+    name='BQWriteDisposition',
+    enum_values=[
+        BQWriteDispositionWriteAppend,
+        BQWriteDispositionWriteEmpty,
+        BQWriteDispositionWriteTruncate,
+    ],
+)
+
+BQEncodingISO_8859_1 = EnumValue(Encoding.ISO_8859_1)
+BQEncodingUTF_8 = EnumValue(Encoding.UTF_8)
+BQEncoding = Enum(name='BQEncoding', enum_values=[BQEncodingISO_8859_1, BQEncodingUTF_8])
+
+BQSourceFormatAvro = EnumValue(SourceFormat.AVRO)
+BQSourceFormatCsv = EnumValue(SourceFormat.CSV)
+BQSourceFormatDatastoreBackup = EnumValue(SourceFormat.DATASTORE_BACKUP)
+BQSourceFormatJson = EnumValue(SourceFormat.NEWLINE_DELIMITED_JSON)
```

Comment:
definitely seems weird to have to put these in global scope. do we have to do that or can we just pass direct to the Enum constructor. our enum definitions don't feel too "pythonic"

Reply:
Yeah good callout, will fix in follow-up commit

Reply:
noting here for future reference: I was worried about having to programmatically reference enum values, but that's not something I'm actually doing anywhere